examples/advanced/processes.ipynb

```diff
@@ -0,0 +1,571 @@
+      "process = nengo.processes.PresentInput(\n",
+      "    inputs, presentation_time=0.1)\n",
+      "\n",
+      "tlen = 0.8\n",
+      "plt.plot(process.trange(tlen), process.run(tlen))\n",
+      "plt.xlim([0, tlen])\n",
+      "plt.ylim([-1, 1]);"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Custom processes\n",
+      "\n",
+      "You can create custom processes\n",
+      "by inheriting from the `nengo.Process` class\n",
+      "and overloading the `make_step` function.\n",
+      "\n",
+      "First, we'll make a simple custom process\n",
+      "that implements a two-dimensional oscillator dynamical system.\n",
+      "The `make_step` function defines a `state` variable\n",
+      "to store the state,\n",
+      "and a fixed `A` matrix that determines\n",
+      "how the state changes over time.\n",
+      "\n",
+      "One advantage to using a process over a simple function\n",
+      "is that if we reset our simulator,\n",
+      "`make_step` will be called again\n",
+      "and the process state\n",
+      "will be restored to the initial state."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "class SimpleOscillator(nengo.Process):\n",
+      "    def make_step(self, shape_in, shape_out, dt, rng):\n",
+      "        A = np.array([[-0.1, -1.], [1., -0.1]])\n",
+      "        state = np.array([1., 0.])\n",
+      "        \n",
+      "        # define the step function, which will be called \n",
+      "        # by the node every time step\n",
+      "        def step(t):\n",
+      "            state[:] += dt * np.dot(A, state)\n",
+      "            return state\n",
+      "        \n",
+      "        return step  # return the step function\n",
+      "    \n",
+      "with nengo.Network() as model:\n",
+      "    a = nengo.Node(SimpleOscillator(), size_in=0, size_out=2)\n",
+      "    a_p = nengo.Probe(a)\n",
+      "    \n",
+      "with nengo.Simulator(model) as sim:\n",
+      "    sim.run(20.0)\n",
+      "    \n",
+      "plt.plot(sim.trange(), sim.data[a_p])\n",
+      "plt.xlabel('time [s]');"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We can generalize this process to one\n",
+      "that can implement arbitrary linear dynamical systems,\n",
+      "given `A` and `B` matrices.\n",
+      "We will overload the `__init__` method\n",
+      "to take and store these matrices,\n",
+      "as well as check the matrix shapes\n",
+      "and set the default size in and out.\n",
+      "The advantage of using the default sizes\n",
+      "is that when we then create a node using the process,\n",
+      "or run the process using `apply`,\n",
+      "we do not need to specify the sizes."
```

Comment:
This is an "advanced" notebook, so hopefully users already know that :) I guess a callout wouldn't hurt, but I don't want to distract from what's going on here, so I'd put it at the end of this example if at all. I'd probably prefer to keep it out for now, but it's not a big deal to me.